### PR TITLE
Pass signore client ID and secret to releaser

### PR DESCRIPTION
**Pull request by mdeggies**
The release job is failing because the signore client ID and secret have not been set in the release job: https://github.com/hashicorp/go-getter/runs/3167243861. These are passed in to the [setup-signore job](https://github.com/hashicorp/go-getter/commit/d4be3502b44c5acc628bdffdf230189cf5e0136b), but it seems they also need to be made available explicitly in the goreleaser context. This PR is meant to address that.

### CODE REVIEW

**Comment by bflad** [comment]
Setting the client id and secret in setup-signore should be creating a configuration file which can be used by later signore invocations, however that step is silently failing with:

```
(node:1577) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/home/runner/.signore/config.yaml'
```

We should update the action to fail loudly for those errors (and of course resolve the issue itself 😄 ). A temporary workflow workaround may be to manually `mkdir $HOME/.signore` before setup-signore.

**Comment by mdeggies** [comment]
Ah, thanks Brian! The secrets can also be set as env vars, right? (Though perhaps we'd need to set more than just the client ID + secret?) :-D

**Comment by milescrabill** [comment]
Yes, the secrets can be set as env vars - `SIGNORE_CLIENT_ID`, `SIGNORE_CLIENT_SECRET`, `SIGNORE_SIGNER`, and more: https://github.com/hashicorp/signore/blob/main/docs/CONFIGURATION.md#client

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -67,6 +67,8 @@ jobs:
           ARTIFACTORY_USER: ${{ secrets.ARTIFACTORY_USER }}
           CIRCLE_TOKEN: ${{ secrets.CIRCLE_TOKEN }}
           SIGNORE_SIGNER: ${{secrets.SIGNORE_SIGNER}}
+          SIGNORE_CLIENT_ID: $${{secrets.SIGNORE_CLIENT_ID}
+          SIGNORE_CLIENT_SECRET: $${{secrets.SIGNORE_CLIENT_SECRET}
```
Comment (bflad) [inline .github/workflows/release.yml:71]: Extra `$` and missing `}` 😎 

```suggestion
          SIGNORE_CLIENT_ID: ${{secrets.SIGNORE_CLIENT_ID}}
          SIGNORE_CLIENT_SECRET: ${{secrets.SIGNORE_CLIENT_SECRET}}
```

**Review by bflad — APPROVED** [review]
Otherwise this should be good to unblock the release process for now. 👍

File: .github/workflows/release.yml (hunk 1 above)
Comment (mdeggies) [reply .github/workflows/release.yml:71]: Oh geez, thanks for catching this Brian. :D

**Review by azr — APPROVED** [review]

**Comment by mdeggies** [comment]
Ok! Merging & running attempt #2. Thanks y'all

**Pull request merged**